docs/en/development/CI.md

```diff
@@ -0,0 +1,209 @@
+## CI checks
+
+When you submit a pull request, some automated checks are ran for your code by
+the ClickHouse [continuous integration (CI) system](tests.md#test-automation).
+This happens after a repository maintainer (someone from ClickHouse team) has
+screened your code and added the `can be tested` label to your pull request.
+The results of the checks are listed on the GitHub pull request page as
+described in the [GitHub checks
+documentation](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-status-checks).
+If a check is failing, you might be required to fix it. This page gives an
+overview of checks you may encounter, and what you can do to fix them.
+
+If it looks like the check failure is not related to your changes, it may be
+some transient failure or an infrastructure problem. Push an empty commit to
+the pull request to restart the CI checks:
+```
+git reset
+git commit --allow-empty
+git push
+```
+
+If you are not sure what to do, ask a maintainer for help.
+
+
+### Merge with master
+
+Verifies that the PR can be merged to master. If not, it will fail with the
+message 'Cannot fetch mergecommit'. To fix this check, resolve the conflict as
+described in the [GitHub
+documentation](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github),
+or merge the `master` branch to your pull request branch using git.
+
+
+### Docs check
+
+Tries to build the ClickHouse documentation website. It can fail if you changed
+something in the documentation. Most probable reason is that some cross-link in
+the documentation is wrong. See the check report and look for `ERROR` messages.
+
+#### Report details
+
+- [Status page example](https://clickhouse-test-reports.s3.yandex.net/12550/eabcc293eb02214caa6826b7c15f101643f67a6b/docs_check.html)
+- `docs_output.txt` contains the building log. [Successful result example](https://clickhouse-test-reports.s3.yandex.net/12550/eabcc293eb02214caa6826b7c15f101643f67a6b/docs_check/docs_output.txt)
+
+
+### Description check
+
+Check that the description of your pull request conforms to the template
+[PULL_REQUEST_TEMPLATE.md](https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md).
+You have to specify a changelog category for your change (e.g., Bug Fix), and
+write a user-readable message describing the change for [CHANGELOG.md](../whats-new/changelog/index.md)
+
+
+### Push to dockerhub
+
+Builds docker images used for build and tests.
+
+
+### Marker check
+
+This check means that the CI system started to process the pull request. When it has 'pending' status, it means that not all checks have been started yet. After all checks have been started, it changes status to 'success'.
+
+
+### Style check
+
+Performs some simple regex-based checks of code style, using the `utils/check-style/check-style` binary.
```

Comment:
add link to it and note that it can be fired locally before pushing the branch https://github.com/ClickHouse/ClickHouse/blob/master/utils/check-style/check-style